core/src/sumeragi/network_topology.rs

```diff
@@ -163,16 +262,39 @@ pub enum Role {
     ProxyTail,
 }
 
+/// Error during signature verification
+#[derive(thiserror::Error, Debug, Clone, Copy, PartialEq, Eq)]
+pub enum SignatureVerificationError {
+    /// Error when block doesn't have enough valid signatures
+    #[error("The block doesn't have enough valid signatures to be committed ({votes_count} out of {min_votes_for_commit}).")]
+    NotEnoughSignatures {
+        /// Current number of signatures
+        votes_count: usize,
+        /// Minimal required number of signatures
+        min_votes_for_commit: usize,
+    },
+    /// Error when block doesn't have proxy tail signature
+    #[error("The block doesn't have proxy tail signature")]
+    ProxyTailMissing,
+    /// Error when block doesn't have leader signature
+    #[error("The block doesn't have leader signature")]
+    LeaderMissing,
+}
+
 #[cfg(test)]
 mod tests {
```

Comment:
Good unit test coverage would have caught the error early. This is why it's important.

Reply:
For each test, I'd refer to the relevant section of the whitepaper so we can verify which parts are to-spec and which are not.